ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -530,11 +552,45 @@ public boolean addBlobStore(ReplicaId replica) {
     logger.info("New store is successfully added into StorageManager");
     return true;
   }
+
+  /**
+   * Build inmemory state for file copy based replication post filecopy is completed.
+   * @param replica the {@link ReplicaId} of the {@link Store} for which store needs to be built
+   */
   @Override
-  public void buildStateForFileCopy(String partitionName){
-    // no-op
+  public void buildStateForFileCopy(ReplicaId replica){
+    if (replica == null) {
+      logger.error("ReplicaId is null");
+      throw new StateTransitionException("ReplicaId null is not found in clustermap for " + currentNode, ReplicaNotFound);
+    }
+    PartitionId partitionId = replica.getPartitionId();
+
+    if (!addBlobStoreForFileCopy(replica)){
```

Comment:
I am confused, I thought we are going to addBlobStoreForFileCopy before the FileCopy finishes all the log files?

Reply:
@justinlin-linkedin As per the design of StateBuild and Bootstrap FileCopy, we have the flow where we we don't addBlobStore before filecopy, but only once state building is triggered. The trigger point for state build will handle latest log and index segment marking active and starting the BlobStore.